ui/src/app/completed/controllers/TableCtrl.js

```diff
@@ -49,17 +65,99 @@
           compressionProgressHandler(response.data['process_id']);
         }else{
           ToastService.showToast('Oops! Something went wrong fetching data');
-          return;
-        }
-        $mdToast.show($scope.zipToast).then(function (toastResponse) {
-        if(toastResponse === 'ok'){
-          ToastService.showToast('Download will start in a while');
         }
+      }, function(error){
+        ToastService.showToast('Oops! Something went wrong fetching data');
       });
+    };
+    $scope.generateSharingLink = function(){
+      if($scope.checkedFileArray.length === 0){
+        ToastService.showToast("Please checkbox some files to share");
+        return;
+      }
+
+      TableService.startCompression($scope.checkedFileArray).then(function (response) {
+        $mdDialog.show(progressDialog);
+        if(response.data['process_id'] != null) {
+          $scope.shareLink = BassaUrl;
+          $scope.shareLink += "/api/file?gid="+response.data['process_id']+"&share=true";
```

Comment:
How long is the proces_id? would someone be able to mine downloads of others by trying out random process_ids?

Reply:
process_id is the combination of gid's which means for every file cluster or single file, it's different.
I have kept the limit of 252 characters. mining gid combination would not be easy